onadata/apps/api/tests/viewsets/test_export_viewset.py

```diff
@@ -622,3 +589,39 @@ def test_export_are_downloadable_to_all_users_when_public_form(self):
         request = self.factory.get("/export", **alices_extra)
         response = self.view(request, pk=export.pk)
         self.assertEqual(response.status_code, 200)
+
+    @patch("onadata.libs.utils.logger_tools.get_storage_class")
+    @patch("onadata.libs.utils.logger_tools.boto3.client")
+    def test_download_from_s3(self, mock_presigned_urls, mock_get_storage_class):
+        """Export is downloaded from Amazon S3"""
+        expected_url = (
+            "https://testing.s3.amazonaws.com/bob/exports/"
+            "trees/csv/trees_2024_06_21_07_47_24_026998.csv?"
+            "response-content-disposition=attachment%3Bfilename%trees.csv&"
+            "response-content-type=application%2Foctet-stream&"
+            "AWSAccessKeyId=AKIAJ3XYHHBIJDL7GY7A"
+            "&Signature=aGhiK%2BLFVeWm%2Fmg3S5zc05g8%3D&Expires=1615554960"
+        )
+        mock_presigned_urls().generate_presigned_url = MagicMock(
+            return_value=expected_url
+        )
+        mock_get_storage_class()().bucket.name = "onadata"
+        self._create_user_and_login()
+        self._publish_transportation_form()
+        export = self._create_export()
+        request = self.factory.get("/export")
+        force_authenticate(request, user=self.user)
+        response = self.view(request, pk=export.pk)
+        self.assertEqual(response.status_code, 302, response.url)
```

Comment:
What does the 3rd argument to `assertEqual` do?

Reply:
@FrankApiyo I copied the assertion from a previous implementation. I will remove it since the 3rd argument should be the message displayed if the assertion fails.